core/domain/collection_domain.py

```diff
@@ -260,39 +254,7 @@ def from_dict(cls, node_dict):
             CollectionNode. The corresponding CollectionNode domain object.
         """
         return cls(
-            copy.deepcopy(node_dict['exploration_id']),
-            copy.deepcopy(node_dict['prerequisite_skill_ids']),
-            copy.deepcopy(node_dict['acquired_skill_ids']))
-
-    @property
-    def skills(self):
-        """Returns a set of skill IDs.
-
-        Returns:
-            set(str). The union of the prerequisite and acquired skill IDs.
-            Each skill is represented at most once.
-        """
-        return set(self.prerequisite_skill_ids) | set(self.acquired_skill_ids)
-
-    def update_prerequisite_skill_ids(self, prerequisite_skill_ids):
-        """Update the prerequise skill IDs.
-
-        Args:
-            prerequisite_skill_ids: list(str). The new list of prerequisite
-                skill IDs to set.
-        """
-
-        self.prerequisite_skill_ids = copy.deepcopy(prerequisite_skill_ids)
-
-    def update_acquired_skill_ids(self, acquired_skill_ids):
-        """Update the acquired skill IDs.
-
-        Args:
-            acquired_skill_ids: list(str). The new list of acquired skill IDs to
-                set.
-        """
-
-        self.acquired_skill_ids = copy.deepcopy(acquired_skill_ids)
+            copy.deepcopy(node_dict['exploration_id']))
```

Comment:
don't think you need the deepcopy, it's just a string

Reply:
Done.